### ISSUE TITLE
exe "hi! User1 cterm=..." does not update the statusline

### ISSUE THREAD

**Issue by Lilja** [vim/vim#1517]
I think I found some weird behavior with setting the User{1-9} highlight group when using the "execute" command. I would like to set the User1 highlight group, dynamically.

Version
```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jan  2 2014 19:39:32)
```

Consider this .vimrc
```
ino <C-C> <ESC>

set laststatus=2

function! Color(mode)
        if (a:mode == "n")
                exe 'hi! User1 ctermbg=green'
        elseif (a:mode == "i")
                exe 'hi! User1 ctermbg=blue'
        elseif (a:mode == "v")
                exe 'hi! User1 ctermbg=white'
        else
                exe 'hi! User1 ctermbg=black'
        endif
        return ''
endfunction

set statusline=%{Color(mode())} " see what mode user is and set appropriate color 
set statusline+=%1*%{mode()} " Show mode in text with User1 highlight group 

" Default User1 green since vim starts usually in normal mode
hi User1 ctermbg=green ctermfg=black
```
**[Here is a gif showcasing the output](https://i.imgur.com/DK3UapS.gifv)**
Textual description: When i'm opening `a`, i'm pressing the sequence `insert(i)`, `normal(C-C)`, `visual(v)`, `visual(v)`. The statusline does not sync with the modes pressed(insert=blue, normal=green, visual=white).  

**Expected output:**
(In _normal_ mode)
        <Insert mode key pressed> => statusline= I with blue background

(In _insert_ mode)
        <Normal mode key pressed> => statusline= N with green background

(In _normal_ mode)
        <visual mode key pressed> => statusline= V with white background

(In _visual_ mode)
        <visual mode key pressed> => statusline= N with green background

**Actual**
(In _normal_ mode)
        <Insert mode key pressed> => statusline= I with blue background

(In _insert_ mode)
        <Normal mode key pressed> => statusline= N with **blue** background

(In _normal_ mode)
        <visual mode key pressed> => statusline= V with **green** background

(In _visual_ mode)
        <visual mode key pressed> => statusline= N with **white** background

Bottom line, it seem like the "hi! User1 ctermfg=whatever" is actually executed for the next mode-press.

**Comment by pudymody** [comment]
i had a similar problem, and when i `set lazyredraw` it worked as expected, dont know why
